### PR TITLE
pep8 and ruff fixes in the plot folder

### PR DESCRIPTION
**Pull request by fchapoton**
This is 
- fixing ruff warning for codes UP and C4
- fixing pep8 code E302
in the `plot` folder

mostly scripted using `ruff` and `autopep8`

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.

### CODE REVIEW

File: src/sage/plot/matrix_plot.py [hunk 1]
```diff
@@ -224,10 +224,10 @@ def _render_on_subplot(self, subplot):
             extent = (lim['xmin'], lim['xmax'],
                       lim['ymax' if flip_y else 'ymin'],
                       lim['ymin' if flip_y else 'ymax'])
-            opts = dict(cmap=cmap, interpolation='nearest', aspect='equal',
-                      norm=norm, vmin=options['vmin'], vmax=options['vmax'],
-                      origin=('upper' if flip_y else 'lower'),
-                      extent=extent, zorder=options.get('zorder'))
+            opts = {'cmap': cmap, 'interpolation': 'nearest', 'aspect': 'equal',
```
Comment (mkoeppe) [inline src/sage/plot/matrix_plot.py]: Do we consider this style change an improvement and why?

Comment (fchapoton) [reply src/sage/plot/matrix_plot.py]: well, this avoids using the `dict` word. Not clear to me if this is really an improvement. I can certainly undo changes in this file.

Comment (mkoeppe) [reply src/sage/plot/matrix_plot.py]: Is this suggestion coming from ruff? Does it have explanations/references for its style codes?

Comment (fchapoton) [reply src/sage/plot/matrix_plot.py]: This is ruff C408 : https://docs.astral.sh/ruff/rules/unnecessary-collection-call/

having to call dict is slower, they say

Comment (mkoeppe) [reply src/sage/plot/matrix_plot.py]: I don't know. Either way is fine with me. But if you want this change, please adjust the indentation

File: src/sage/plot/plot_field.py [hunk 2]
```diff
@@ -355,6 +355,6 @@ def plot_slope_field(f, xrange, yrange, **kwds):
         norm_inverse = lambda x,y: 1/sqrt(f(x, y)**2+1)
         f_normalized = lambda x,y: f(x, y)*norm_inverse(x, y)
     else:
-        norm_inverse = 1 / sqrt((f**2+1))
+        norm_inverse = 1 / sqrt(f**2+1)
```
Comment (mkoeppe) [inline src/sage/plot/plot_field.py]:
```suggestion
        norm_inverse = 1 / sqrt(f**2 + 1)
```

File: src/sage/plot/plot.py [hunk 3]
```diff
@@ -2066,13 +2069,14 @@ def f(x): return (floor(x)+0.5) / (1-(x-0.5)**2)
             xmax = kwds.pop('xmax', 1)
             G = _plot(funcs, (xmin, xmax), *args, **kwds)
         else:
-            sage.misc.verbose.verbose("there were %s extra arguments (besides %s)" % (n, funcs), level=0)
+            sage.misc.verbose.verbose("there were {} extra arguments (besides {})".format(n, funcs), level=0)
```
Comment (mkoeppe) [inline src/sage/plot/plot.py]:
```suggestion
            sage.misc.verbose.verbose(f"there were {n} extra arguments (besides {funcs})", level=0)
```

File: src/sage/plot/colors.py [hunk 4]
```diff
@@ -338,7 +338,7 @@ def rgbcolor(c, space='rgb'):
 
     elif isinstance(c, (list, tuple)):
         if len(c) != 3:
-            raise ValueError("color list or tuple '%s' must have 3 entries, one for each RGB, HSV, HLS, or HSL channel" % (c, ))
+            raise ValueError("color list or tuple '{}' must have 3 entries, one for each RGB, HSV, HLS, or HSL channel".format(c ))
```
Comment (mkoeppe) [inline src/sage/plot/colors.py]:
```suggestion
            raise ValueError(f"color list or tuple '{c}' must have 3 entries, one for each RGB, HSV, HLS, or HSL channel")
```

File: src/sage/plot/arc.py [hunk 5]
```diff
@@ -346,7 +346,7 @@ def _repr_(self):
             sage: print(Arc(2,3,2.2,2.2,0,2,3,{}))
             Arc with center (2.0,3.0) radii (2.2,2.2) angle 0.0 inside the sector (2.0,3.0)
         """
-        return "Arc with center (%s,%s) radii (%s,%s) angle %s inside the sector (%s,%s)" % (self.x, self.y, self.r1, self.r2, self.angle, self.s1, self.s2)
+        return "Arc with center ({},{}) radii ({},{}) angle {} inside the sector ({},{})".format(self.x, self.y, self.r1, self.r2, self.angle, self.s1, self.s2)
```
Comment (mkoeppe) [inline src/sage/plot/arc.py]: Why not go the next step already and make it f-strings.
Same comment applies to some other files below

File: src/sage/plot/plot3d/plot3d.py [hunk 6]
```diff
@@ -348,14 +348,14 @@ def to_cartesian(self, func, params=None):
...
-                                     dep_var_dummy=dep_var_dummy,
-                                     indep_var_dummies=indep_var_dummies))
+                ll = f"""lambda {params[0]},{params[1]}: t.subs({{
+                    dep_var_dummy: float(func({params[0]}, {params[1]})),
+                    indep_var_dummies[0]: float({params[0]}),
+                    indep_var_dummies[1]: float({params[1]})
+                }})"""
+                return eval(ll, {'t': t, 'func': func,
+                                     'dep_var_dummy': dep_var_dummy,
+                                     'indep_var_dummies': indep_var_dummies})
```
Comment (mkoeppe) [inline src/sage/plot/plot3d/plot3d.py]:
```suggestion
                return eval(ll, {'t': t, 'func': func,
                                 'dep_var_dummy': dep_var_dummy,
                                 'indep_var_dummies': indep_var_dummies})
```

File: src/sage/plot/graphics.py [hunk 7]
```diff
@@ -1437,37 +1436,37 @@ def _set_scale(self, subplot, scale=None, base=None):
     # NOTE: If you intend to use a new parameter in show(), you should update
     # this dictionary to contain the default value for that parameter.
 
-    SHOW_OPTIONS = dict(# axes options
-                        axes=None, axes_labels=None, axes_labels_size=None,
-                        axes_pad=None, base=None, scale=None,
-                        xmin=None, xmax=None, ymin=None, ymax=None,
-                        flip_x=False, flip_y=False,
+    SHOW_OPTIONS = {# axes options
+                        'axes': None, 'axes_labels': None, 'axes_labels_size': None,
```
Comment (mkoeppe) [inline src/sage/plot/graphics.py]: indentation needs fixing in this file

**Review by mkoeppe — APPROVED** [review]
LGTM.

**Comment by vbraun** [comment]
merge conflict

**Comment by fchapoton** [comment]
back to positive after trivial merge

**Pull request merged**